Platforms/QemuQ35Pkg/CpuInfoDxe/CpuInfoDxe.c

```diff
@@ -0,0 +1,129 @@
+/** @file CpuInfoDxe.c
+
+    Copyright (c) Microsoft Corporation. All rights reserved.
+
+    Driver that prints CPU branding information
+**/
+
+#include <Uefi.h>
+#include <Library/UefiLib.h>
+#include <Library/DebugLib.h>
+#include <Library/UefiDriverEntryPoint.h>
+#include <Library/UefiBootServicesTableLib.h>
+#include <Library/BaseLib.h>
+#include <Library/PrintLib.h>
+#include <Library/UefiCpuLib.h>
+
+typedef union {
+  ///
+  /// Individual bit fields
+  ///
+  struct {
+    UINT32    SteppingId       : 4; ///< [Bits   3:0] Stepping ID
+    UINT32    Model            : 4; ///< [Bits   7:4] Model
+    UINT32    FamilyId         : 4; ///< [Bits  11:8] Family
+    UINT32    ProcessorType    : 2; ///< [Bits 13:12] Processor Type
+    UINT32    Reserved1        : 2; ///< [Bits 15:14] Reserved
+    UINT32    ExtendedModelId  : 4; ///< [Bits 19:16] Extended Model ID
+    UINT32    ExtendedFamilyId : 8; ///< [Bits 27:20] Extended Family ID
+    UINT32    Reserved2        : 4; ///< Reserved
+  } Bits;
+  ///
+  /// All bit fields as a 32-bit value
+  ///
+  UINT32    Uint32;
+} CPUID_VERSION_INFO_EAX;
+
+#define CPUID_VERSION_INFO   0x01
+#define CPUID_BRAND_STRING1  0x80000002
+#define CPUID_BRAND_STRING2  0x80000003
+#define CPUID_BRAND_STRING3  0x80000004
+#define MAX_MESSAGE_LENGTH   64
+
+STATIC CHAR8  mMessage[MAX_MESSAGE_LENGTH];
+
+CHAR8 *
+EFIAPI

```

Comment:
If these functions are only called within the module (from the entry point), you can drop `EFIAPI`.